graph_shortest_path/routing.py

```diff
@@ -52,8 +54,23 @@ def bfs(self, start):
         @param {Vertex} start: The starting vertex
         """
         # !!!! IMPLEMENT ME
-        pass
-
+        for vertex in self.vertices:
+            vertex.color = 'white'
+            vertex.parent=  None
+
+            start.color = 'grey'
+            queue = [start]
+            while queue:
+                current = queue.pop(0)
+
+                for edge in current.edges:
+                    neighbor = edge.destination
+                    if neighbor.color == 'white':
+                        neight.color = 'grey'
+                        neight.parent = current
+                        queue.append(neighbor)
+
+                current.color = black
```

Comment:
consider de-indenting this somehow...you have a for inside a while inside a for.......